src/meta/src/stream/source_manager.rs

```diff
@@ -1033,6 +1038,46 @@ impl SourceManager {
         for source_id in source_ids {
             if let Some(handle) = core.managed_sources.remove(&source_id) {
                 handle.handle.abort();
+                // remove the shared kafka client entry if it exists
+                if let Some(entry) = handle.share_client_entry {
```

Comment:
Is there any reason for manually maintaining the ref count? Does simply storing a `Weak` in the cache work?